packages/workspace/src/workspace/alias_index.ts

```diff
@@ -0,0 +1,131 @@
+/*
+*                      Copyright 2023 Salto Labs Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import {
+  Change,
+  getChangeData,
+  Element,
+  toChange,
+  CORE_ANNOTATIONS,
+  isAdditionChange,
+  isModificationChange,
+  isObjectType, ObjectType, isObjectTypeChange, ModificationChange,
+} from '@salto-io/adapter-api'
+import { logger } from '@salto-io/logging'
+import _ from 'lodash'
+import { prettifyName } from '@salto-io/adapter-utils'
+import { ElementsSource } from './elements_source'
+import { getAllElementsChanges } from './index_utils'
+import { RemoteMap } from './remote_map'
+
+
+const log = logger(module)
+export const ALIAS_INDEX_VERSION = 4
+const ALIAS_INDEX_KEY = 'alias_index'
+
+const getFieldsElemIDsFullName = (objectType: ObjectType): string[] =>
+  Object.values(objectType.fields).map(field => field.elemID.getFullName())
+
+const getChangedFields = (change: ModificationChange<ObjectType>): Change<Element>[] => {
+  const afterFields = Object.values(change.data.after.fields)
+  const beforeFields = Object.values(change.data.before.fields)
+  const additionFields = afterFields.filter(field => !beforeFields.find(f => f.elemID.isEqual(field.elemID)))
+  const removalFields = beforeFields.filter(field => !afterFields.find(f => f.elemID.isEqual(field.elemID)))
+  const aliasModificationFields = afterFields
+    .filter(afterField => {
+      const beforeField = beforeFields.find(f => f.elemID.isEqual(afterField.elemID))
```

Comment:
can be 
```suggestion
      const beforeField = change.data.before.fields[afterField.name]
```
(but - then we should check `isField(beforeField)` and not just `!== undefined`, to avoid some edge cases with builtin functions)